src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -457,7 +468,8 @@ class FunSpec private constructor(
     internal val String.isConstructor get() = this == CONSTRUCTOR
     internal val String.isAccessor get() = this.isOneOf(GETTER, SETTER)
 
-    private val EXPRESSION_BODY_PREFIX = CodeBlock.of("return ")
+    private val RETURN_EXPRESSION_BODY_PREFIX = CodeBlock.of("return ")
+    private val THROW_EXPRESSION_BODY_PREFIX = CodeBlock.of("throw")
```

Comment:
Adding a space to the end of the prefix is important to avoid cases like this:
```kotlin
val funSpec = FunSpec.builder("foo")
    .returns(Int::class)
    .addStatement("throwsAFoo()")
    .addStatement("return 1 + 1")
    .build()
```
If the space is omitted, the output will be incorrect:
```kotlin
expected:
    fun foo(): kotlin.Int {
      throwsAFoo()
      return 1 + 1
    }
    
but was:
    fun foo(): kotlin.Int = throwsAFoo()
    return 1 + 1
```
Space here basically ensures that `throw` is matched only when it's used as a keyword.

Reply:
Yup, good point! 
02ae906ac6a7deded0da407a271ee8a5cedd1205